apps/wallet/src/ui/app/staking/stake/ValidatorDetailCard.tsx

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { isSuiObject, isSuiMoveObject, SUI_TYPE_ARG } from '@mysten/sui.js';
+import { useMemo } from 'react';
+
+import { getName, STATE_OBJECT } from '../usePendingDelegation';
+import { Card } from '_app/shared/card';
+import CoinBalance from '_app/shared/coin-balance';
+import { ImageIcon } from '_app/shared/image-icon';
+import Alert from '_components/alert';
+import LoadingIndicator from '_components/loading/LoadingIndicator';
+import { useGetObject, useAppSelector } from '_hooks';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+import { Text } from '_src/ui/app/shared/text';
+import { IconTooltip } from '_src/ui/app/shared/tooltip';
+
+import type { ValidatorState } from '../ValidatorDataTypes';
+import type { ReactNode } from 'react';
+
+function SplitList({ children }: { children: ReactNode[] }) {
+    return <div className="flex py-2.5 gap-2 items-center">{children}</div>;
+}
+
+export function ValidateDetailFormCard({
+    validatorAddress,
+    unstake,
+}: {
+    validatorAddress: string;
+    unstake?: boolean;
+}) {
+    const accountAddress = useAppSelector(({ account }) => account.address);
+    const { data, isLoading, isError } = useGetObject(STATE_OBJECT);
+
+    const validatorsData =
+        data && isSuiObject(data.details) && isSuiMoveObject(data.details.data)
+            ? (data.details.data.fields as ValidatorState)
+            : null;
+
+    const validatorDataByAddress = useMemo(() => {
```

Comment:
If I get it right this is the validator's data (transformed) for the selected address. I would use one memo to find the validator by address `validatorsData.validators.fields.active_validators.find(...)` and then based on that either use memos for the other values or just extract them inline. I don't think map is necessary here